crates/optimism/chainspec/src/lib.rs

```diff
@@ -232,7 +239,8 @@ impl EthChainSpec for OpChainSpec {
     }
 
     fn genesis_header(&self) -> &Self::Header {
-        self.inner.genesis_header()
+        // header is set in construction of `OpChainSpec` from `ChainSpec`
+        self.inner.genesis_header.get().expect("header is set")
```

Comment:
the `From<ChainSpec>` impl ensures that this is safe here. the visibility of the inner type is restricted to private to ensure that the chainspec is constructed using the conversion from `ChainSpec`

Reply:
there was no more obvious place to set the l2 withdrawals field, the semantics of the original code weren't straight forward

Reply:
> semantics of the original code weren't

which ones?

all I'm saying is that's unclear to me where the actual change is